shapely/algorithms/oriented_envelope.py

```diff
@@ -0,0 +1,136 @@
+import math
+from itertools import islice
+from typing import Optional, Union
+
+import numpy as np
+
+import shapely
+from shapely import oriented_envelope_geos
+from shapely.affinity import affine_transform
+
+__all__ = [
+    "oriented_envelope_min_area",
+    "oriented_envelope",
+    "minimum_rotated_rectangle",
+]
+
+
+def oriented_envelope_min_area(geometry, **kwargs):
+    """
+    Computes the oriented envelope (minimum rotated rectangle) that encloses an input geometry.
+
+    Refer to `shapely.oriented_envelope` with `method=min_area` for full documentation.
+    """
+    if geometry is None:
+        return None
+    if not hasattr(geometry, "geom_type"):
+        return np.array([oriented_envelope_min_area(g) for g in geometry])
```

Comment:
I'm not sure what are suggesting to change here.
This inline should be faster than `np.vectorize` since it doesn't have the generalization overhead of `np.vectorized` which states `The [vectorize](https://numpy.org/doc/stable/reference/generated/numpy.vectorize.html#numpy.vectorize) function is provided primarily for convenience, not for performance. The implementation is essentially a for loop.`

Reply:
All the shapely ufuncs provide the full functionality of ufuncs, which among other things is that it works on arrays of any dimension, not just on 1D arrays. 
I made my comment above in the assumption that your code would only work for 1D arrays (seeing the single for loop), but missed that this is of course working recursively, and so will therefore also work on >1D arrays (the outer loop slices out one dimension, until you iterate over scalars).

So my comment wasn't for performance considerations, I know that in the end it's always a python loop. But numpy provides some helpers that turn a python function that works on scalar values into a fully compatible ufunc function. So it could still be useful to use that (but it might be `np.frompyfunc` and not `np.vectorize` that we want, not fully sure about the differences between both)

Reply:
> So my comment wasn't for performance considerations, I know that in the end it's always a python loop. But numpy provides some helpers that turn a python function that works on scalar values into a fully compatible ufunc function. So it could still be useful to use that (but it might be `np.frompyfunc` and not `np.vectorize` that we want, not fully sure about the differences between both)

On this specific case, that there is only single parameter to vectorized - the numpy wrappers wouldn't add any advantages. in general,  `np.vectorize` wraps `np.frompyfunc` to add even more generalized options (like auto deduction of input/output types), thus slower. So I don't think we should use either of them in this specific case, where we only iterate on a single argument.